src/test/relation_engine_client/test_integration.py

```diff
@@ -0,0 +1,137 @@
+import unittest
+import os
+from uuid import uuid4
+
+from src.relation_engine_client import REClient
+from src.relation_engine_client.exceptions import RERequestError, RENotFound
+
+_API_URL = os.environ.get('RE_API_URL', 'http://localhost:5000')
+_VERT_COLL = 'test_vertex'
+_EDGE_COLL = 'test_edge'
+_TOK_ADMIN = 'admin_token'
+_TOK_USER = 'non_admin_token'
+_TOK_INVALID = 'invalid_token'
+
+
+class TestREClientIntegration(unittest.TestCase):
+    """Integration tests for the REClient package."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.client = REClient(_API_URL, _TOK_ADMIN)
+        pass
+
+    def test_admin_query_ok(self):
+        _id = self._save_test_vert()
+        bind_vars = {'id': _id}
+        query = f"FOR vert IN {_VERT_COLL} FILTER vert._key == @id RETURN vert"
+        result = self.client.admin_query(query, bind_vars)
+        self.assertEqual(result['count'], 1)
+        self.assertEqual(result['results'][0]['_key'], _id)
+
+    def test_admin_empty_query(self):
+        bind_vars = {'id': 'xyz'}
+        with self.assertRaises(RERequestError) as ctx:
+            self.client.admin_query("", bind_vars)
+        self.assertEqual(ctx.exception.resp.status_code, 400)
+
+    def test_admin_missing_param(self):
+        query = f"FOR vert IN {_VERT_COLL} FILTER vert._key == @id RETURN vert"
+        with self.assertRaises(RERequestError) as ctx:
+            self.client.admin_query(query, bind_vars={})
+        self.assertEqual(ctx.exception.resp.status_code, 400)
+
+    def test_admin_raise_not_found(self):
+        query = f"FOR vert IN {_VERT_COLL} FILTER vert._key == @id RETURN vert"
+        _id = str(uuid4())
+        bind_vars = {'id': _id}
+        with self.assertRaises(RENotFound) as ctx:
+            self.client.admin_query(query, bind_vars, raise_not_found=True)
+        self.assertTrue(_id in ctx.exception.req_body)
+
+    def test_admin_invalid_args(self):
+        # No params
+        with self.assertRaises(TypeError):
+            self.client.admin_query()
+        # Wrong type for query
+        with self.assertRaises(TypeError):
+            self.client.admin_query(123)
```

Comment:
I don't understand how this test passes. AFAIK python doesn't enforce type hints itself, and there's nothing in the admin method I can see that'd cause a type error (e.g. trying to use the number as a string)

Reply:
Also applies to similar tests

Reply:
Also curious why this works. Checking it out

Reply:
Two of the TypeErrors are raised due to python throwing "required positional arguments" errors.

The last type error is `123 is not iterable`, raised when trying to cast `123` as a dict in order to use it as bind_vars in the query.

I'll add explicit runtime type checks, because why not? Won't slow anything down in this case.